rest/utilities_testing.go

```diff
@@ -2310,13 +2308,50 @@ func WaitAndAssertBackgroundManagerExpiredHeartbeat(t testing.TB, bm *db.Backgro
 
 // RespRevID returns a rev ID from the given response, or fails the given test if a rev ID was not found.
 func RespRevID(t *testing.T, response *TestResponse) (revID string) {
+	revision := DocVersionFromPutResponse(t, response)
+	return revision.RevID
+}
+
+// DocVersion represents a specific version of a document in an revID/HLV agnostic manner.
+type DocVersion struct {
+	DocID string
```

Comment:
My first instinct would have been to omit DocID from this, and have it just include RevID and (in future) currentVersion.  My feeling is that this actually obfuscates what's going on in a helper function like DocUpdate, and it's better to keep docID and DocVersion as separate parameters to that kind of method.

What do you think the benefits are of including and requiring docID in this struct?

Reply:
The benefits were because this is often used together and it makes sense to store it conceptually.

```
doc1Version := rt.PutDoc("doc1", `{"a" : true}`)
rt.WaitForVersion(doc1Version)
doc1Version = rt.UpdateDoc(doc1Version, `{"a" : false}"`)
rt.WaitForVersion(doc1Version)
rt.DeleteDoc(doc1Version)
```

vs:

```
doc1ID := "doc1"
doc1Version := rt.PutDoc(doc1ID, `{"a" : true}`)
rt.WaitForVersion(doc1ID, doc1Version)
doc1Version = rt.UpdateDoc(doc1ID, doc1Version, `{"a" : false}"`)
rt.WaitForVersion(doc1ID, doc1Version)
rt.DeleteDoc(doc1ID, doc1Version)
```

I'm not settled on this way of doing things.

Reply:
I prefer the latter.  We need to define docID anyway (for PutDoc in the first example), and I think it keeps the kv API (PutDoc, UpdateDoc, etc) more consistent to always take docID and then an optional version.